File: Insertion Sort/Insertion-sort-Go/InsertionSort.go
package main

import "fmt"

// custom type array
type MyArray []int64

func (arr MyArray) InsertionSort() {
	n := len(arr)
	for i := 1; i < n; i++ {
		key, j := arr[i], i-1
		for j >= 0 && arr[j] > key {
			arr[j+1] = arr[j]
			j--
		}
		arr[j+1] = key

	}
}

func main() {
	arr := MyArray{12, 11, 13, 5, 6}
	fmt.Printf("Before sort: %v\n", arr) //Before sort: [12 11 13 5 6]
	arr.InsertionSort()
	fmt.Printf("After sort: %v\n", arr) //After sort: [5 6 11 12 13]
}

File: Insertion Sort/Insertion-sort-Matlab/insertion_sort.m
function [array] = insertion_sort(array)
	for i = 2:array.size()
		tmp = array[i];
		j = i - 1;
		while (j => 1 && array[j] > tmp)
			array[j+1] = array[j];
			j = j-1;
		end

		array[j+1] = tmp;
	end
end
